pkg/client/client.go

```diff
@@ -325,6 +325,11 @@ func (c *KpmClient) Compile(kclPkg *pkg.KclPkg, kclvmCompiler *runner.Compiler)
 		kclvmCompiler.AddDepPath(dName, dPath)
 	}
 
+	kclPkg.UpdatePkgEdition(kclvmCompiler.GetKclVersion())
+	if err := kclPkg.ModFile.StoreModFile(); err != nil {
```

Comment:
Remove the function `kclPkg.ModFile.StoreModFile` calling.

Reply:
done! 
I have made changes according to https://github.com/kcl-lang/kpm/issues/186#issuecomment-1950987112